File: wc.rb
#!/usr/bin/ruby

txt = ARGV[0]

puts "Ruby"
printf("Processing %s ...\n", txt)

fp = File.open(txt)

count = 0
wc = Hash.new { |h,k| h[k] = 0 }

fp.each do |line|
    count += 1
    #line.split(/ |\.|,|;|\-/).each { |word|
    line.split().each do |word|
        next if word == "\n"
        wc[word] += 1
    end
end

fp.close()

printf("Processed %d lines.\n", count)

#wc.sort_by{|k,v|v}.reverse().each{ |k, v| printf("%10d: [%s]\n", v, k) }

File: wc.py
#!/usr/bin/env python3

from sys import argv
from collections import defaultdict

txt = argv[1]

print("Python")
print("Processing", txt, "...")

fp = open(txt)

count = 0
wc = defaultdict(int)

for line in fp:
    #print(line, )
    count += 1

    # Remove \n and blank line$
    #line = line.rstrip("\n")
    #if (len(line) == 0):
    #    continue

    for word in line.split():
        wc[word] += 1

fp.close()

print("Processed", count, "lines.")

for word, value in sorted(wc.items(), key=lambda item: item[1], reverse=True):
    print("%10d: [%s]" % (value, word))
